model/RepositoryService.php

```diff
@@ -178,9 +180,22 @@ public function restore(Revision $revision)
             $this->getModel()->getRdfInterface()->add($triple);
         }
 
+        $item = $this->getQtiService()->getDataItemByRdfItem($resource);
```

Comment:
Regarding @jbout comment about extracting this line from here. 
What do you think about that?

Reply:
you can , if he feels that this is more elegant